go tests fail: `TestNoRTMPListener` and `TestServerDisconnectMidStream`

blocks #136
```$ go test ./...
?   	github.com/livepeer/lpms/cmd/example	[no test files]
?   	github.com/livepeer/lpms/cmd/example/simple	[no test files]
?   	github.com/livepeer/lpms/cmd/transcoding	[no test files]
ok  	github.com/livepeer/lpms/core	(cached)
ok  	github.com/livepeer/lpms/ffmpeg	(cached)
HandShake: client signature does not match!
[hls @ 0x901ee00] Cannot use rename on non file protocol, this may lead to races and temporary partial files
WriteN, RTMP send error 32 (5 bytes)
WriteN, RTMP send error 32 (42 bytes)
RTMP_ReadPacket, failed to read RTMP packet body. len: 5604
HandShake: client signature does not match!
WriteN, RTMP send error 32 (5 bytes)
WriteN, RTMP send error 32 (42 bytes)
RTMP_ReadPacket, failed to read RTMP packet body. len: 5604
No application or playpath in URL!
RTMP_Connect0, failed to connect socket. 61 (Connection refused)
segmenter: Unable to open input
--- FAIL: TestNoRTMPListener (0.00s)
    video_segmenter_test.go:419: error was not nil; got  Unknown error occurred
HandShake: client signature does not match!
RTMP_ReadPacket, failed to read RTMP packet header
[flv @ 0xa002400] Could not find codec parameters for stream 0 (Video: h264, none): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
--- FAIL: TestServerDisconnectMidStream (0.11s)
    video_segmenter_test.go:464: Expected 'Input/output error' but instead got  <nil>
FAIL
FAIL	github.com/livepeer/lpms/segmenter	3.606s
ok  	github.com/livepeer/lpms/stream	(cached)
ok  	github.com/livepeer/lpms/transcoder	(cached)
ok  	github.com/livepeer/lpms/vidlistener	(cached)
ok  	github.com/livepeer/lpms/vidplayer	(cached)
```

--- a/cmd/example/simple/simple.go
+++ b/cmd/example/simple/simple.go
@@ -13,6 +13,12 @@ import (
 	"github.com/livepeer/lpms/stream"
 )
 
+type exampleStream string
+
+func (t exampleStream) StreamID() string {
+	return string(t)
+}
+
 func randString(n int) string {
 	rand.Seed(time.Now().UnixNano())
 	x := make([]byte, n, n)
@@ -34,9 +40,9 @@ func main() {
 	lpms := core.New(&core.LPMSOpts{WorkDir: fmt.Sprintf("%v/.tmp", dir)})
 
 	lpms.HandleRTMPPublish(
-		func(url *url.URL) (strmID string) {
+		func(url *url.URL) stream.AppData {
 			glog.Infof("Stream has been started!: %v", url)
-			return randString(10)
+			return exampleStream(randString(10))
 		},
 
 		func(url *url.URL, rs stream.RTMPVideoStream) (err error) {

